### ISSUE TITLE
Server error when searching

### ISSUE THREAD

**Issue by amartinkyle** [Cairath/tools-not-included#59]
**Describe the bug**
Server error while searching like screenshot. Have tried 15 times and consistently get error.  Is this public source code so i could debug?

![oni](https://user-images.githubusercontent.com/81704425/209417938-6c15dd04-a20d-4769-9d4e-5ff3bd419ff2.png)

**Comment by Cairath** [comment]
No it is not and will not be. Reduce the number of filters, the server's having issues with the load and unfortunately i don't have the time to work on it any soon.

**Comment by yuxincs** [comment]
Would you consider posting a snapshot of the database for us to use locally?

**Comment by chris9871** [comment]
Hi @Cairath ! I love the site and would like to help.  Is there a way I could assist?

**Comment by MarisKori** [comment]
Sometimes even 2-3 filters cause error.

**Comment by MarisKori** [comment]
For example, these filters just threw an error:
![image](https://user-images.githubusercontent.com/18173966/210446945-07244448-8b60-410e-a70e-bce47124ef6d.png)
...
![image](https://user-images.githubusercontent.com/18173966/210447241-28c731d0-f35f-4280-9ab8-18a1d99c0218.png)

**Comment by Art94** [comment]
If we are not allowed to fix it "from the inside", we will make life easier for ourselves "from the outside" ))))

var jq = document.createElement('script');
jq.src = "https://ajax.googleapis.com/ajax/libs/jquery/3.5.1/jquery.min.js";
document.getElementsByTagName('head')[0].appendChild(jq);
// ... give time for script to load, then type (or see below for non wait option)
jQuery.noConflict();

setInterval(function (){
if($("button > span:contains('Return to the Main Page')").parent().length)
	{
	$("button > span:contains('Return to the Main Page')").parent().click()
	$("button > span:contains('Search')").parent().click()
	}
}, 2000);

**Comment by Art94** [comment]
am glad that this comment provoked a response) The app still "almost didn't work". Why not publish at least a snapshot of the database, I do not understand. I really feel sorry for the time of everyone who installed the seed transfer mod (including mine), to create "your personal" database). By the way, I have plenty of ip addresses, it's better to spend time fixing errors

**Comment by Cairath** [comment]
I don't deal with entitled assholes. The site's been up and running for years while i don't even play the game anymore.  If i had the time to fix it now, i would. I've been providing a free service, so you can feel sorry for yourself all you want - I don't. Vanilla searches work fine, SO is struggling a lot. But sure, thanks to jerks like you I'll just take it down.

**Comment by Cairath** [comment]
I found some time to try to put some optimizations in place. Deployed a new version this morning, will see how it does.

**Comment by Stuyvenstein** [comment]
Hi Cairath. Anything more than 2 filters on Spaced Out map searches still results in the above server error.

**Comment by Amselfass** [comment]
Hi Cairath! I unfortunately cannot get any Spaced Out map browser searches to work, even if I only choose a single filter. They all result in the server error screen already shown above.

I realize that you no longer play this game and have little time to spend on fixing the site, and that is of course acceptable. Would you perhaps be willing to give some more consideration to the previous suggestion of making the database available for download, in whatever export format is most conveniant for you?

**Comment by Cairath** [comment]
It's not about fixing anymore at this point - I can't throw more money at the problem, the server's basically maximized on resources and prices for hosting are skyrocketing year by year. I've sunk a lot of cash into this project but I just cannot justify spending more.
The database is over 15GB at this point, it's not a reasonable download.

I considered gutting the filtering and removing  a lot of complexity - as that is something that would actually help, but with recent klei's announcement about adding configurable starts, with all the new story traits, mods to configure the clusters/asteroids -- I'm actually considering putting the project to rest. I feel like it's served its purpose, mostly in vanilla when all geysers were a huge random mess. When configurable starts become a thing, the complexity will be just unfeasible to support - and it's already getting a bit ridiculous with the story traits.

**Comment by manuelgr0** [comment]
Unfortunately, I still get the error when searching for maps. The first search with 3 parameters worked but all the subsequent searches with 0-5 parameters are not working anymore. 
I was searching for Spaced Out + Terra maps.

In case you need help with development / optimizing the search, I'm happy to help. Provided you are willing to share the source code.
Otherwise, it would also be great if you are willing to share some raw data of you database with us.

![image](https://user-images.githubusercontent.com/16121649/235131624-cda49acd-f8d2-460f-9eab-c5fc5f0ecaa6.png)

**Comment by tim-sebring** [comment]
Cairath, for what it's worth, I appreciated (and still do) all the effort and money you've put into this project. I used the tool for a long time to create a new "ideal" map for whatever playstyle I wanted to do, and it was amazing work. Too bad for all the entitled a-holes as you said, but just know there are those of us out here that really appreciate what you did.

**Comment by amartinkyle** [comment]
> Cairath, for what it's worth, I appreciated (and still do) all the effort and money you've put into this project. I used the tool for a long time to create a new "ideal" map for whatever playstyle I wanted to do, and it was amazing work. Too bad for all the entitled a-holes as you said, but just know there are those of us out here that really appreciate what you did.

I dont think anyone is trying to be an a-hole. We obviously all love what was created. It actually kind of spoiled us, having a feature that isnt/wasnt in the game. There is no audible tone when text is just on a screen, but i think most people on here want to help.

**Comment by Xaekai** [comment]
Cairath, we've talked on Discord before.

I would be more than happy to offer serious hosting resources, gratis and in perpetuity, for this project to continue operating, until such time as its actually obsolete by Klei provided customization features. Which at the time of this writing, does not seem to be the case.

**Comment by chris9871** [comment]
Xaekai (and Cairath), I'd be happy to help out with the coding side since this is no longer an active project for Cairath.

**Comment by igorsantos07** [comment]
@Cairath, besides the (reasonable) ideas of leaving the database to download on Google Drive and open-sourcing an almost defunct codebase so we can revive it... Another option is OS-ing the mod itself, so it could store its details on a CSV or SQLite instead of uploading to the dead and costly server, and then we can manually query whatever we were able to generate during the night :)

Best 💕
